client/src/auth/index.js

```diff
@@ -5,10 +5,13 @@ import config from '../config'
 const { apiUrl } = config
 
 export const [useAuth, authFetch, login, logout] = createAuthProvider({
-  onUpdateToken: (token) => {
-    fetch(`${apiUrl}/auth/refresh`, {
+  onUpdateToken: async (oldToken) => {
+    const res = await fetch(`${apiUrl}/auth/refresh`, {
       method: 'POST',
-      body: token.access_token,
-    }).then((r) => r.json())
+      body: oldToken,
+    })
+    const data = await res.json()
+
+    return Promise.resolve(data['access_token'])
```

Comment:
`onUpdateToken` expects a Promise as a return value, so wrapping the return value in a Promise https://github.com/obabichev/react-token-auth/blob/master/src/index.ts#L170